services/net/indexing/IndexingManager.cs

```diff
@@ -457,17 +467,31 @@ private async Task SendNotifications(IndexRequestModel request, ContentModel con
     {
         this.Logger.LogDebug($"SendNotifications:BEGIN:{content.Id}");
 
-        if (!String.IsNullOrWhiteSpace(this.Options.NotificationTopic))
+        var cacheKey = $"{CacheKeyPrefix}{content.Id}";
+
+        if (!_cache.TryGetValue(cacheKey, out _))
         {
-            // TODO: Make request to API to determine what notifications should be sent.
-            // TODO: Generate appropriate notification request.
-            var notification = new NotificationRequestModel(NotificationDestination.SignalR | NotificationDestination.NotificationService, content.Id)
+            if (!String.IsNullOrWhiteSpace(this.Options.NotificationTopic))
             {
-                RequestorId = request.RequestorId
-            };
-            _ = await this.Api.SendMessageAsync(notification)
-                ?? throw new HttpClientRequestException($"Failed to receive result from Kafka when sending message.  Topic: {this.Options.NotificationTopic}, Content ID: {content.Id}");
+                // TODO: Make request to API to determine what notifications should be sent.
+                // TODO: Generate appropriate notification request.
+                var notification = new NotificationRequestModel(NotificationDestination.SignalR | NotificationDestination.NotificationService, content.Id)
+                {
+                    RequestorId = request.RequestorId
+                };
+                _ = await this.Api.SendMessageAsync(notification)
+                    ?? throw new HttpClientRequestException($"Failed to receive result from Kafka when sending message.  Topic: {this.Options.NotificationTopic}, Content ID: {content.Id}");
+
+                // add to cache
+                _cache.Set(cacheKey, true, TimeSpan.FromSeconds(CacheExpirationSecond));
```

Comment:
The problem is this is a memory sink bug.  It'll continue to grow slowly until the pod crashes.

Reply:
You will need to find a solution where these variables don't keep increasing, but rather are cleared.

Reply:
yes, you are right. could has a better way.